haproxy/datadog_checks/haproxy/haproxy.py

```diff
@@ -107,8 +107,9 @@ def check(self, instance):
             info, data = self._fetch_socket_data(parsed_url)
             self._collect_version_from_socket(info)
         else:
-            self._collect_version_from_http(url)
-            data = self._fetch_url_data(url)
+            http_url = "%s%s" % (url, STATS_URL)
+            self._collect_version_from_http(http_url)
+            data = self._fetch_url_data(http_url)
```

Comment:
From this comment https://github.com/DataDog/integrations-core/blob/c7e6bbe97a06c78f0514b6d56406d72dfd087f92/haproxy/datadog_checks/haproxy/haproxy.py#L185-L186

Seems the version call was intended to use the html version.

Reply:
Should we just protect the call to get the version in case of failure?

```
        if parsed_url.scheme == 'unix' or parsed_url.scheme == 'tcp':
            info, data = self._fetch_socket_data(parsed_url)
            self._collect_version_from_socket(info)
        else:
            try:
                self._collect_version_from_http(url)
            except Exception as e:
                self.log.warning("Error collecting version info: %s", e)
            data = self._fetch_url_data(url)
```